tests/test_db.py

```diff
@@ -303,7 +305,8 @@ def test_optimistic_locking(node_factory, bitcoind):
     with pytest.raises(RpcError, match=r'Connection to RPC server lost.'):
         l1.rpc.newaddr()
 
-    assert(l1.daemon.is_in_log(r'Optimistic lock on the database failed'))
```

Comment:
We shouldn't switch stream that we report errors on, people might be relying on the current behavior, e.g., not collect `stderr`. See the link-time specialization I mentioned 3 commits earlier.

Reply:
Agreed, this was definitely sub-optimal! (Though I will say that having the errors print to stderr made debugging problems while developing much much faster ;))